DOCS-2385: add interface verification step for BPF dataplane configuration

ivansharamok opened the pull request:
In some corner cases cluster nodes may have custom interface names which may not get matched by the `bpfDataIfacePattern` regex. Need to clarify what users need to change to have their nodes BPF dataplane ready.

Product Version(s):
CE v3.19+

Issue:
[DOCS-2385](https://tigera.atlassian.net/browse/DOCS-2385)

Link to docs preview:


SME review:
- [ ] An SME has approved this change. 

DOCS review:
- [ ] A member of the docs team has approved this change.

Additional information:
This needs to be added to all versions of Calico docs.

Merge checklist:
- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed 


[DOCS-2385]: https://tigera.atlassian.net/browse/DOCS-2385?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

ivansharamok committed:
add interface verification step for BPF dataplane configuration

In some corner cases cluster nodes may have custom interface names which may not get matched by the `bpfDataIfacePattern` regex. Need to clarify what users need to change to have their nodes BPF dataplane ready.

ctauchen commented:
Given that this is an edge case, I wonder whether this wouldn't fit better in the [ebpf troubleshooting page](https://docs.tigera.io/calico/latest/operations/ebpf/troubleshoot-ebpf)?

ivansharamok commented:
this is not necessarily an edge-case scenario. Interface names on the nodes differ based on the environment. For instance, the type of a VM provided by cloud providers can have different interface names, VMWare sets its own naming pattern, any virtualization solution can use its own naming pattern, on-prem bare-metals can have their own naming patterns. This is a step that everyone should do to make sure their cluster is configured correctly before they switch to Calico BPF mode.

ivansharamok committed:
update node interface naming section

adjust the node interface naming pattern section according to feedback received from the engineering team.

ivansharamok commented:
adjusted the message in the proposed section to verify the node's interface naming pattern according to the feedback received from the engg team.

ctauchen commented:
LGTM, will follow up with changes to later versions.
